src/algebra/fields/bigint.hpp

```diff
@@ -50,6 +50,17 @@ class bigint {
     void to_mpz(mpz_t r) const;
     bool test_bit(const std::size_t bitno) const;
 
+    template<mp_size_t m> inline bigint<n> operator+(const bigint<m>& other) const;
```

Comment:
The ``template<mp_size_t n>`` is implicit because it's declared on the class. So it is consistent, in the strange world of C++.